cpp/include/cudf/null_mask.hpp

```diff
@@ -97,4 +98,50 @@ cudf::size_type count_set_bits(bitmask_type const* bitmask, size_type start,
 cudf::size_type count_unset_bits(bitmask_type const* bitmask, size_type start,
                                  size_type stop);
 
+/**---------------------------------------------------------------------------*
+ * @brief Creates a `device_buffer` from a slice of bitmask defined by a range
+ * of indices `[begin_bit, end_bit)`.
+ *
+ * Returns empty `device_buffer` if `bitmask == nullptr`.
+ *
+ * @throws `cudf::logic_error` if `begin_bit > end_bit`
+ * @throws `cudf::logic_error` if `begin_bit < 0`
+ *
+ * @param mask Bitmask residing in device memory whose bits will be copied
+ * @param begin_bit Index of the first bit to be copied (inclusive)
+ * @param end_bit Index of the last bit to be copied (exclusive)
+ * @param stream Optional, stream on which all memory allocations and copies
+ * will be performed
+ * @param mr Optional, the memory resource that will be used for allocating
+ * the device memory for the new device_buffer
+ * @return rmm::device_buffer A `device_buffer` containing the bits
+ * `[begin_bit, end_bit)` from `mask`.
+ *---------------------------------------------------------------------------**/
+rmm::device_buffer copy_bitmask(
+    bitmask_type const * mask, size_type begin_bit, size_type end_bit,
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**---------------------------------------------------------------------------*
+ * @brief Copies `view`'s bitmask from the bits
+ * `[view.offset(), view.offset() + view.size())` into a `device_buffer`
+ *
+ * Returns empty `device_buffer` if the column is not nullable
+ *
+ * @throws `cudf::logic_error` if `begin_bit > end_bit`
+ * @throws `cudf::logic_error` if `begin_bit < 0`
+ *
+ * @param view Column view whose bitmask needs to be copied
+ * @param begin_bit Index of the first bit to be copied (inclusive)
+ * @param end_bit Index of the last bit to be copied (exclusive)
+ * @param stream Optional, stream on which all memory allocations and copies
+ * will be performed
+ * @param mr Optional, the memory resource that will be used for allocating
+ * the device memory for the new device_buffer
+ * @return rmm::device_buffer A `device_buffer` containing the bits
+ * `[view.offset(), view.offset() + view.size())` from `view`'s bitmask.
+ *---------------------------------------------------------------------------**/
+rmm::device_buffer copy_bitmask(column_view const& view, cudaStream_t stream,
```

Comment:
@kaatish , why were `stream` and `mr` not given default values? Can I add them or was there some issue?

Reply:
Good point, this api should be in `detail` with defaulted stream/mr.

Reply:
@devavret That was an oversight. Yes, it should be updated.